business_requirement_deliverable_project/models/business.py

```diff
@@ -33,12 +35,76 @@ def generate_projects_wizard(self):
         compute='_compute_task_count'
     )
 
+    total_hour = fields.Float(
+        string='Total Hours in Timesheets related to business requirement',
+        compute='_compute_hour'
+    )
+
+    total_planned_hour = fields.Float(
+        string='Total Planned Hour in RL related to business requirement',
+        compute='_compute_planned_hour'
+    )
+
+    @api.multi
+    def write(self, vals):
+        for r in self:
+            if vals.get('state'):
+                ir_obj = self.env['ir.model.data']
+                br_xml_id = ir_obj.\
+                    get_object('business_requirement',
+                               'group_business_requirement_manager')
+                user = self.env['res.users']
+                grps = [grp.id for grp in user.browse(self._uid).groups_id]
+                date = fields.Datetime.now()
+                if vals['state'] == 'confirmed':
+                    vals.update({'confirmed_id': user,
+                                 'confirmation_date': date})
+                if vals['state'] == 'draft':
+                    vals.update({'confirmed_id': False,
+                                 'approved_id': False,
+                                 'confirmation_date': False,
+                                 'approval_date': False
+                                 })
+                if vals['state'] == 'approved':
+                    if br_xml_id.id in grps:
+                        vals.update({'approved_id': user,
+                                     'approval_date': date})
+                    else:
+                        raise Warning('You can only move to the following\
+                            stage: draft/confirmed/cancel/drop.')
+                if vals['state'] in ('stakeholder_approval', 'in_progress',
+                                     'done'):
+                    if br_xml_id.id not in grps:
+                        raise Warning('You can only move to the following\
+                            stage: draft/confirmed/cancel/drop.')
+            return super(BusinessRequirement, self).write(vals)
```

Comment:
(rant) This is how it can be done right now, but in the future I wish we have a base module/library to handle Stage/State approvals and allowed changes...